Diggy/PressureMap.lua

```diff
@@ -58,25 +64,26 @@ end
     @return Table with maxed_values_buffer, quadrant1, quadrant2, quadrant3 and quadrant4
 ]]
 local function get_pressure_map(surface)
-    if (nil == pressure_map_storage[surface.name]) then
-        pressure_map_storage[surface.name] = {
-            -- contains all coordinates that are at max pressure until cleared.
-            maxed_values_buffer = {},
-
-            -- map with coordinates, stored as [x][y] = fraction.
-            quadrant1 = {},
-            quadrant2 = {},
-            quadrant3 = {},
-            quadrant4 = {},
-        }
+    if not global.pressure_map_storage[surface.index] then
+
+      global.pressure_map_storage[surface.index] = {}
+
+      local map = global.pressure_map_storage[surface.index]
+
+      map.quadrant1 = setmetatable({},_mt_x)
+      map.quadrant2 = setmetatable({},_mt_x)
+      map.quadrant3 = setmetatable({},_mt_x)
+      map.quadrant4 = setmetatable({},_mt_x)
+
+      map.maxed_values_buffer = {}
     end
 
-    return pressure_map_storage[surface.name]
+    return global.pressure_map_storage[surface.index]
 end
 
 function PressureMap.process_maxed_values_buffer(surface, callback)
-    if ('table' ~= type(surface) or nil == surface.name) then
-        error('PressureMap.process_maxed_values_buffer argument #1 expects a LuaSurface, ' .. type(surface) .. ' given.')
+    if ('table' ~= type(surface) or not surface.name) then
+        error('PressureMap.process_maxed_values_buffer argument #1 expects a LuaSurface, ' ..surface.name .. ' given.')

```

Comment:
Sorry for touching that, didnt mean to i just fiddled with it. But i think it still does the same thing.
(not name) == (nil == name)

Or am i wrong about that?

Reply:
Old condition: If it's not a table or surface has no name -> print the type of the variable
new condition: If it's not a table or surface as no name -> print nil/false (because just asserted it didn't exist)